Добавлена переменная MutationAmplitude

Nematin opened the pull request:
Добавлена переменная MutationAmplitude в метод Mutate.
Изменён класс GeneticAlgorithm

Nematin committed:
Добавлена переменная MutationAmplitude

Добавлена переменная MutationAmplitude в метод Mutate.
Изменён класс GeneticAlgorithm

Mikhail-Vaganov commented on GeneticConsole/GeneticAlgorithm.cs:
```diff
@@ -21,18 +21,19 @@ public class GeneticAlgorithm
         public double[] BestGenes { get; set; }
         public int PopulationSize { get; set; }
-
+        public double MutationAmplitude { get; set; }
```
Эта переменная относится к особи, а не к генетическому алгоритму же. У тебя правильно метод мутации лежит в соответствующем поле класса Specimen, т.к. про то, как мутировать, должны знать именно особи. Этот же класс GeneticAlgorithm управляет лишь общим алгоритмом.
Так что перенеси поле в Specimen.

Mikhail-Vaganov commented on GeneticConsole/GeneticAlgorithm.cs:
```diff
@@ -73,7 +74,7 @@ public void NewGeneration(double crossoverRate)
                     Specimen child = firstParent.Crossover(secondParent);
                     if (random.NextDouble() < MutationRate)
-                        child.Mutate(child.Genes);
+                        child.Mutate(child.Genes, MutationAmplitude);
```
смотри, как странно выглядит: получается, что мы действительно каким-то внешним фактором настраиваем мутацию при каждом вызове этого метода

Nematin replied:
Исправил, теперь mutationAmplitude в классе Specimen

Nematin replied:
ЧТо то пошло не по плану, теперь у меня конфликт веток

Nematin committed:
Update GeneticTestEnvironment.cs

Nematin committed:
Update GeneticTestEnvironment.cs

Nematin committed:
Upadate

Nematin committed:
Update

Nematin committed:
Изменена фит функция

Теперь на вход метода вместо индекса особи подаётся сама особь.

С индексом мне было понятнее

Nematin committed:
Merge branch 'master' into test
